parsers/MD.py

```diff
@@ -1,103 +1,96 @@
-# Data for the fuel-mix at CERS Moldovenească for the year 2020 can be found at:
-# http://moldgres.com/wp-content/uploads/2021/02/proizvodstvennye-pokazateli-zao-moldavskaja-gres-za-2020-god.pdf
-
-# Annual reports from moldelectrica can be found at:
-# https://moldelectrica.md/ro/network/annual_report
-
-
-def template_price_response(zone_key: str, datetime: datetime, price) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "currency": "MDL",
-        "price": price,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_consumption_response(
-    zone_key: str, datetime: datetime, consumption
-) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "consumption": consumption,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_production_response(zone_key: str, datetime: datetime, production) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "production": production,
-        "storage": {},
-        "source": "moldelectrica.md",
-    }
-
-
-def template_exchange_response(
-    sorted_zone_keys: str, datetime: datetime, netflow
-) -> dict:
-    return {
-        "sortedZoneKeys": sorted_zone_keys,
-        "datetime": datetime,
-        "netFlow": netflow,
-        "source": "moldelectrica.md",
-    }
-
-
-def get_archive_data(
+ARCHIVE_BASE_URL = "https://moldelectrica.md/utils/archive2.php?id=table&type=array"
+SOURCE = "moldelectrica.md"
+
+
+# Moldoelectrica electricity tariffs as defined by government-agency decisions.
+_MOLDOELECTRICA_NEW_POWER_PRICE_IN_MDL_PER_MW = {
+    # https://www.legis.md/cautare/getResults?doc_id=78826&lang=ro
+    datetime(2000, 4, 1, tzinfo=timezone.utc): 18.8,
+    # https://www.legis.md/cautare/getResults?doc_id=103953&lang=ro
+    datetime(2001, 10, 1, tzinfo=timezone.utc): 28.0,
+    # https://www.legis.md/cautare/getResults?doc_id=40249&lang=ro
+    datetime(2002, 9, 1, tzinfo=timezone.utc): 35.2,
+    # https://www.legis.md/cautare/getResults?doc_id=42701&lang=ro
+    datetime(2005, 9, 1, tzinfo=timezone.utc): 39.3,
+    # https://www.legis.md/cautare/getResults?doc_id=10948&lang=ro
+    datetime(2007, 8, 3, tzinfo=timezone.utc): 51.8,
+    # https://www.legis.md/cautare/getResults?doc_id=40130&lang=ro
+    datetime(2010, 1, 19, tzinfo=timezone.utc): 63.2,
+    # https://www.legis.md/cautare/getResults?doc_id=40589&lang=ro
+    datetime(2012, 5, 11, tzinfo=timezone.utc): 80.2,
+    # https://www.legis.md/cautare/getResults?doc_id=84436&lang=ro
+    datetime(2015, 7, 31, tzinfo=timezone.utc): 145.0,
+    # https://www.legis.md/cautare/getResults?doc_id=134854&lang=ro
+    datetime(2023, 12, 31, tzinfo=timezone.utc): 201.0,
+    # https://www.legis.md/cautare/getResults?doc_id=142391&lang=ro
+    datetime(2024, 3, 21, tzinfo=timezone.utc): 185.0,
+    # TODO(amv213): update as new tariffs get rolled out...
+}
```

Comment:
Are these constant? As in they don't change hourly?

And do you know if they are just the electricity price or the total cost (including transmission fees and taxes)?

Reply:
Yes, these are constants. It looks like each ['licence holder' has a regulated price](https://www.anre.md/energie-electrica-3-290), and every so often the National Agency for Energy Regulation of the Republic of Moldova (ANRE) emits new decrees to update them to new ones (e.g. like [here](https://www.legis.md/cautare/getResults?doc_id=142391&lang=ro) for the latest one). Each decree abrogates the previous one, so I tried to walk them all back one by one to try to cover all historical changes.

I think that the price includes transmission but not taxes: [here](https://www.anre.md/energie-electrica-3-290) they refer to it as "Tariff/Regulated price money/kWh (without VAT)", and both the ANRE decrees and [moldoelectrica](https://moldelectrica.md/ro/activity/tariff) themselves refer to it as "tariful pentru serviciul de transport al energiei electrice (TVA excluded)" which seems to be translating as "Tariff for the electricity transmission service"

Reply:
(The previous code was essentially always returning the `datetime(2015, 7, 31, tzinfo=timezone.utc): 145.0` price)

Reply:
Thanks!

Everything looks good then.